### PR TITLE
docs: add contribution guidelines

### PR DESCRIPTION
**Pull request by chrisvxd**

### CODE REVIEW

**Review by monospaced — APPROVED** [review]
Made a bunch of comments, but nothing that I feel requires amendment. Happy for this to be merged as-is, but leaving my notes here in case they are helpful.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+## Labels
+
+We manage our backlog using labels. Labels can help you understand the status of each ticket.
+
+### Status labels
+
+- **ready** - this ticket has a description and is ready to be worked on.
+- **in triage** - this ticket has been seen by the Puck team and we are identifying next steps. Tickets may stay in this state until we're ready to process them.
+- **opinions wanted** - we're looking for opinions on this ticket. Feel free to chime in with comments or suggestions.
```
[the remark this answers is not in the document]
Comment (chrisvxd) [reply CONTRIBUTING.md:26]: 1. I think mostly. Bug fixes might also require some opinions. Also sometimes it's softer, like "how do you want to engage with puck".
2. Probably not, but I don't think it needs to be _that_ strict.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,120 @@
+# Contribution guidelines
+
+Puck is still under heavy development, having gained significant interest at an early stage. The contribution guidelines are designed to help us balance community engagement with our vision and direction.
+
+## Reporting bugs or requesting features
+
+Bugs and feature requests are best reported via GitHub issues. If you're unsure whether or not you've encounter a bug, feel free to ask first.
```
[the remark this answers is not in the document]
Comment (chrisvxd) [reply CONTRIBUTING.md:7]: Did think that but we don't have the label yet... will check the backlog and make one if I think there are any issues

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,120 @@
+# Contribution guidelines
```
[the remark this answers is not in the document]
Comment (anglepoised) [reply CONTRIBUTING.md]: Or just "Contributing to Puck"

File: CONTRIBUTING.md (hunk 2 above)
Comment (anglepoised) [inline CONTRIBUTING.md]:
```suggestion
Bugs and feature requests are best reported via GitHub issues. If you're unsure whether or not you've encountered a bug, feel free to ask first.
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,126 @@
+# Puck contributing guide
+
+Puck is still under heavy development, having gained significant interest at an early stage. The contribution guidelines are designed to help us balance community engagement with our vision and direction.
+
+## Reporting bugs or requesting features
+
+Bugs and feature requests are best reported via GitHub issues. Always check if the issue already exists before opening a new one.
+
+If you're unsure whether or not you've encounter a bug, feel free to ask first 👇
```
Comment (anglepoised) [inline CONTRIBUTING.md:9]:
```suggestion
If you're unsure whether or not you've encountered a bug, feel free to ask first 👇
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+
+## Asking questions
+
+There are several ways to ask questions or ask for help:
+
+- Open a [discussion](https://github.com/measuredco/puck/discussions) via GitHub
+- Use the #chat or #help channels in our [Discord server](https://discord.gg/D9e4E3MQVZ)
```
Comment (anglepoised) [inline CONTRIBUTING.md:16]: Maybe less friction for people to point to just one Discord channel? 

```suggestion
- Use the #help channel in our [Discord server](https://discord.gg/D9e4E3MQVZ)
```

File: CONTRIBUTING.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (chrisvxd) [reply CONTRIBUTING.md]: Added!

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+- Open a discussion via GitHub
+- Use the #chat or #help channels in our [Discord server](https://discord.gg/D9e4E3MQVZ)
+
+**GitHub issues are for bugs and feature requests, not for questions.**
+
+## Labels
+
+We manage our backlog using labels. Labels can help you understand the status of each ticket.
+
+### Status labels
```
[the remark this answers is not in the document]
Comment (chrisvxd) [reply CONTRIBUTING.md:24]: Added!

File: CONTRIBUTING.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (chrisvxd) [reply CONTRIBUTING.md]: Updated!

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+If you're unsure whether or not you've encounter a bug, feel free to ask first 👇
+
+## Asking questions
+
+There are several ways to ask questions or ask for help:
+
+- Open a [discussion](https://github.com/measuredco/puck/discussions) via GitHub
+- Use the #chat or #help channels in our [Discord server](https://discord.gg/D9e4E3MQVZ)
+
+_GitHub issues are for bugs and feature requests, not for questions._
```
Comment (anglepoised) [inline CONTRIBUTING.md:18]: Friendly reword.

```suggestion
_Please only use GitHub issues for bugs and feature requests, and not for questions._
```

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+Bugs and feature requests are best reported via GitHub issues. If you're unsure whether or not you've encounter a bug, feel free to ask first.
+
+## Asking questions
+
+There are several ways to ask questions or ask for help:
+
+- Open a discussion via GitHub
+- Use the #chat or #help channels in our [Discord server](https://discord.gg/D9e4E3MQVZ)
+
+**GitHub issues are for bugs and feature requests, not for questions.**
```
[the remark this answers is not in the document]
Comment (chrisvxd) [reply CONTRIBUTING.md]: Updated!

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -0,0 +1,126 @@
...
+
+- **type: bug**
+- **type: feature**
+- **type: docs**
+- **type: performance**
+- **type: test**
+
+### Other labels
+
+- **good first issue** - if you're new to contributing on Puck, this is a good palce to start.
```
Comment (anglepoised) [inline CONTRIBUTING.md:42]:
```suggestion
- **good first issue** - if you're new to contributing on Puck, this is a good place to start.
```

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -0,0 +1,126 @@
...
+#### CSS
+
+- Class names must follow [SUIT CSS](https://suitcss.github.io) convention. This is a tooling-angostic convention used at [@measuredco](https://github.com/measuredco) for all CSS work.
+- Don't rely on global styles. Puck is deployed into hostile third-party environments and we have no control over what CSS may be running on the page.
+
+#### Commits
+
+**Keep your PRs focused to a single issue**. This makes it easier to review and is necessary for our release process.
+
+We rely on angular-style conventional commits for automating our releases, determining the version bump and generating our changelog.
```
Comment (anglepoised) [inline CONTRIBUTING.md:106]: Worth linking to a guide to Angular-style, or just Conventional Commits docs here?

File: CONTRIBUTING.md [hunk 11]
```diff
@@ -0,0 +1,126 @@
...
+
+You generally don't need to write perfect commit messages yourselves - we squash most PRs and rewrite the messages on merge.
+
+If you need to solve multiple issues, it's best to split it into multiple PRs. Or, if you're comfortable writing convential commits, you can also split each change into a separate commit. The team is more likely to have opinions about this and you may be asked to reword your commits.
```
Comment (anglepoised) [inline CONTRIBUTING.md:110]:
```suggestion
If you need to solve multiple issues, it's best to split it into multiple PRs. Or, if you're comfortable writing conventional commits, you can also split each change into a separate commit. The team is more likely to have opinions about this and you may be asked to reword your commits.
```

File: CONTRIBUTING.md [hunk 12]
```diff
@@ -0,0 +1,126 @@
...
+
+PRs without issues may be accepted for small fixes, but larger changes may be rejected or require further discussion.
+
+### Setting up the environment
+
+Puck uses:
+
+- TypeScript
+- CSS Modules
+- Turbo repo for efficient monorepo scripting
```
Comment (anglepoised) [inline CONTRIBUTING.md:71]:
```suggestion
- Turborepo for efficient monorepo scripting
```

File: CONTRIBUTING.md [hunk 13]
```diff
@@ -0,0 +1,126 @@
...
+### Style
+
+#### TypeScript
+
+- Avoid the use of `any`.
+- Tests appreciated, but not required. They may be requested for complicated code.
+
+#### CSS
+
+- Class names must follow [SUIT CSS](https://suitcss.github.io) convention. This is a tooling-angostic convention used at [@measuredco](https://github.com/measuredco) for all CSS work.
```
Comment (anglepoised) [inline CONTRIBUTING.md:99]:
```suggestion
- Class names must follow [SUIT CSS](https://suitcss.github.io) convention. This is a tooling-agnostic convention used at [@measuredco](https://github.com/measuredco) for all CSS work.
```

**Review by anglepoised — CHANGES_REQUESTED** [review]

**Pull request merged**